### PR TITLE
argocd: 2.3.2 -> 2.3.3

### PR DESCRIPTION
**Pull request by bryanasdev000**
###### Description of changes

Version bump, also adding backport label to fix security issues (https://github.com/argoproj/argo-cd/security/advisories/GHSA-2f5v-8r3f-8pww)

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by bryanasdev000** [comment]
TODO: Remove commit and change reference to src.rev.

File: pkgs/applications/networking/cluster/argocd/default.nix [hunk 1]
```diff
@@ -2,20 +2,17 @@
 
 buildGoModule rec {
   pname = "argocd";
-  version = "2.3.2";
+  version = "2.3.3";
   tag = "v${version}";
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/argocd/default.nix:6]:
```suggestion
```

File: pkgs/applications/networking/cluster/argocd/default.nix [hunk 2]
```diff
@@ -2,20 +2,17 @@
...
-  # Update commit to match the tag above
-  # TODO make updadeScript
-  commit = "ecc2af9dcaa12975e654cde8cbbeaffbb315f75c";
 
   src = fetchFromGitHub {
     owner = "argoproj";
     repo = "argo-cd";
     rev = tag;
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/argocd/default.nix:11]:
```suggestion
    rev = "v${version}";
```

File: pkgs/applications/networking/cluster/argocd/default.nix [hunk 3]
```diff
@@ -27,7 +24,7 @@ buildGoModule rec {
       "-s" "-w"
       "-X ${package_url}.version=${version}"
       "-X ${package_url}.buildDate=unknown"
-      "-X ${package_url}.gitCommit=${commit}"
+      "-X ${package_url}.gitCommit=${src.rev}"
       "-X ${package_url}.gitTag=${tag}"
```
Comment (zowoq) [inline pkgs/applications/networking/cluster/argocd/default.nix:28]:
```suggestion
      "-X ${package_url}.gitTag=${src.rev}"
```

**Review by zowoq — COMMENTED** [review]
Could drop `tag` as well.

**Comment by bryanasdev000** [comment]
> Backport failed for `release-21.11`, because it was unable to cherry-pick the commit(s).
> 
> Please cherry-pick the changes locally.
> 
> ```shell
> git fetch origin release-21.11
> git worktree add -d .worktree/backport-167299-to-release-21.11 origin/release-21.11
> cd .worktree/backport-167299-to-release-21.11
> git checkout -b backport-167299-to-release-21.11
> ancref=$(git merge-base d240ca09aa43b3650da8b5b5ebc77053e7ecbaf6 a18a398b17a6eeac2000507beeb4c7be5ec344c0)
> git cherry-pick -x $ancref..a18a398b17a6eeac2000507beeb4c7be5ec344c0
> ```

Since we are close to a new release, I will not be backporting it anymore.

**Pull request merged**